core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java

```diff
@@ -142,11 +158,22 @@ public AuthenticationToken getToken() {
       }
     }
 
-    if (securePassword != null)
-      return new PasswordToken(securePassword.value);
-
-    if (password != null)
-      return new PasswordToken(password.value);
+    // other token types should have resolved by this point, so return PasswordToken
+    Password pass = null;
+    if (securePassword != null) {
+      pass = securePassword;
+    } else if (password != null) {
+      pass = password;
+    } else {
+      try {
+        pass = Password.promptUser();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    if (pass != null) {
```

Comment:
OK I like removing the null return. If I get rid of the try/catch, I'd have to make getToken() throw IOException which will touch a bunch of files.